libs/state/spec/transformation-helpers/array/toDictionary.spec.ts

```diff
@@ -0,0 +1,106 @@
+
+const dictionaryByNumber = {
+  1: {id: 1, type: 'cat', real: true, breeds: ['Persian']},
+  2: {id: 2, type: 'dog', real: true, breeds: ['Doberman']},
+  3: {id: 3, type: 'catDog', real: false, breeds: []}
+};
+
+const dictionaryByString = {
+  cat: {id: 1, type: 'cat', real: true, breeds: ['Persian']},
+  dog: {id: 2, type: 'dog', real: true, breeds: ['Doberman']},
+  catDog: {id: 3, type: 'catDog', real: false, breeds: []}
+};
+
+const dictionaryBySymbol = {
+  felis: {id: 1, type: 'cat', real: true, breeds: ['Persian'], [genus]: 'felis'},
+  canis: {id: 2, type: 'dog', real: true, breeds: ['Doberman'], [genus]: 'canis'},
+  fake: {id: 3, type: 'catDog', real: false, breeds: [], [genus]: 'fake'}
+}
+
+beforeEach(() => {
+  creatures = [
+    {id: 1, type: 'cat', real: true, breeds: ['Persian']},
+    {id: 2, type: 'dog', real: true, breeds: ['Doberman']},
+    {id: 3, type: 'catDog', real: false, breeds: []}
+  ];
+});
+
+
+describe('toDictionary', () => {
+
+  describe('general', () => {
+    it('should be defined', () => {
+      const fn = toDictionary;
+
+      expect(fn).toBeDefined();
+    });
+  });
+
+  describe('functionality', () => {
+    it('should create dictionary by number', () => {
+      const dictionaryResult = toDictionary(creatures, 'id');
+
+      expect(dictionaryResult).toEqual(dictionaryByNumber);
+    });
+
+    it('should create dictionary by string', () => {
+      const dictionaryResult = toDictionary(creatures, 'type');
+
+      expect(dictionaryResult).toEqual(dictionaryByString);
+    });
+
+    it('should create dictionary by symbol', () => {
+      const dictionaryResult = toDictionary(Object.values(dictionaryBySymbol), genus);
+
+      expect(dictionaryResult).toEqual(dictionaryBySymbol);
+    });
+  });
+
+  describe('edge cases', () => {
+    it('should work with empty initial array', () => {
+      const dictionaryResult = toDictionary([] as any, 'fakeKey');
+
+      expect(dictionaryResult).toEqual({});
+    });
+
+    it('should return empty object if key does not exist', () => {
+      expect(toDictionary(creatures, 'fake' as any)).toEqual({});
+    });
+
+    it('should return empty object when key value is not string, number or symbol', () => {
+      expect(toDictionary(creatures, 'breeds' as any)).toEqual({});
+    });
+
+    it('should return empty object when key is not provided', () => {
+      expect(toDictionary(creatures, null as any)).toEqual({});
+    });
+
+    it('should return empty object when first argument is not array', () => {
+      expect(toDictionary('' as any, '')).toEqual({});
+      expect(toDictionary({} as any, '')).toEqual({});
+      expect(toDictionary(1 as any, '')).toEqual({});
+      expect(toDictionary(false as any, '')).toEqual({});
+    });
+
+    it('should return null or undefined when array is not provided', () => {
+      const arr: any[] = null as any;
+      const arr2: any[] = undefined as any;
+      expect( toDictionary(arr, '')).toEqual(undefined);
```

Comment:
shouldn't this return null?

Reply:
Maybe yes. But select doing filter(slice => slice !== undefined). So idea was to return undefined even in case of null.

Overkill?

Reply:
Fixed